kotlin/samples/recyclerview/src/main/java/com/squareup/sample/recyclerview/editablelistworkflow/ListDiffMode.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.recyclerview.editablelistworkflow
+
+import com.squareup.workflow.ui.ContainerHintKey
+
+/**
+ * Tells [EditableListAdapter]s how to calculate and notify updates.
+ */
+enum class ListDiffMode {
+
+  None,
+  Synchronous,
+  Asynchronous;
+
+  companion object : ContainerHintKey<ListDiffMode>(ListDiffMode::class) {
+    override val default get() = None
```

Comment:
will it play nice with Java, if someone will try to mix and match languages?

Reply:
It will look weird, but I don't see any reason why it wouldn't work. But Java support is not something we really care about.